src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -2964,10 +2964,11 @@ private RefKind GetEffectiveParameterRefKind(ParameterSymbol parameter, RefKind
                 }
                 else
                 {
+                    bool isThisArgument = arguments.IsExtensionMethodThisArgument(argumentPosition);
                     RefKind parameterRefKind = parameters.ParameterRefKinds.IsDefault ? RefKind.None : parameters.ParameterRefKinds[argumentPosition];
-                    conversion = CheckArgumentForApplicability(candidate, argument, argumentRefKind, parameters.ParameterTypes[argumentPosition], parameterRefKind, ignoreOpenTypes, ref useSiteDiagnostics);
+                    conversion = CheckArgumentForApplicability(candidate, argument, argumentRefKind, parameters.ParameterTypes[argumentPosition], parameterRefKind, ignoreOpenTypes, isThisArgument, ref useSiteDiagnostics);
 
-                    if (arguments.IsExtensionMethodThisArgument(argumentPosition) && !Conversions.IsValidExtensionMethodThisArgConversion(conversion))
+                    if (isThisArgument && !Conversions.IsValidExtensionMethodThisArgConversion(conversion))
```

Comment:
>ExtensionMethodThisArgConversion [](start = 62, length = 32)

`ref` this should not allow ThisArg conversions except for identity conversions. 
It may already be inforce in CheckArgumentForApplicability, but we need tests for that like:
- pass (int, int)  to ref readonly (object, object), should work
- pass (int, int) to ref (object, object), should not work
- pass (int, int) to ref (int Alice, int Bob), should work